pkg/apis/core/v1/helper/helpers.go

```diff
@@ -413,6 +413,33 @@ func TolerationsTolerateTaintsWithFilter(tolerations []v1.Toleration, taints []v
 	return true
 }
 
+// FindMatchingUntoleratedTaint checks if the given tolerations tolerates
+// all the filtered taints, and returns the first taint without a toleration
+func FindMatchingUntoleratedTaint(taints []v1.Taint, tolerations []v1.Toleration, applyFilter taintsFilterFunc) *v1.Taint {
+	filteredTaints := getFilteredTaints(taints, applyFilter)
+	for _, taint := range filteredTaints {
+		if !TolerationsTolerateTaint(tolerations, &taint) {
+			return &taint
+		}
+	}
+	return nil
+}
+
+// getFilteredTaints returns a list of taints satisfying the filter predicate
+func getFilteredTaints(taints []v1.Taint, applyFilter taintsFilterFunc) []v1.Taint {
+	filteredTaints := []v1.Taint{}
+	if applyFilter == nil {
```

Comment:
This should be the first thing in the function, and then you don't need to check `applyFilter != nil` in line 435.

Reply:
changed

Reply:
changed it